### DISCUSSION TITLE
SQL query at server start with rocket_db_pools?

### DISCUSSION THREAD

**Discussion by antimeme** [rwf2/Rocket#2402] [Questions] [answered]
I've got a server using SQLite with rocket_db_pools.  I'd like to issue some SQL queries before the server is started (mainly CREATE TABLE IF NOT EXISTS so that my server can initialize its own schema -- ideally I'd like to create the SQLite file if doesn't exist too).  Is there a way to grab a connection from a pool before or shortly after ignition?

Here's what I imagine the solution might look like:

```
#[derive(rocket_db_pools::Database)]
#[database("serverdb")]
struct ServerDB(sqlx::SqlitePool);

fn create_server() -> rocket::Rocket<rocket::Build> {
    let db = ServerDB::init();

    // do something clever with db here?
    // sqlx::query("CREATE TABLE IF NOT EXISTS ...")
    //       .execute(db.get_connection()).await;

    rocket::build()
        .attach(db)
        .mount("/", /* ... routes that use the database ... */)
}
```

I've looked for answers that might address this, but the closest I can find is #2326.  The answer there doesn't seem to involve sqlx pools or even a rocket instance.  Am I the only one who wants to execute queries before the server starts?

**Comment by SergioBenitez** [comment, marked as answer]
The database example does exactly this: https://github.com/SergioBenitez/Rocket/blob/master/examples/databases/src/sqlx.rs.

**Reply by antimeme** [reply]
Thanks!  I've been reading the guide but didn't know this example existed.
